pkg/provider/onepassword/onepassword.go

```diff
@@ -0,0 +1,364 @@
+	errExpectedOneVault                   = "expected one 1Password Vault matching %w"
+	errExpectedOneItem                    = "expected one 1Password Item matching %w"
+	errGetItem                            = "error finding 1Password Item: %w"
+	errKeyNotFound                        = "key not found in 1Password Vaults: %w"
+	errDocumentNotFound                   = "error finding 1Password Document: %w"
+	errExpectedOneField                   = "expected one 1Password ItemField matching %w"
+
+	documentCategory = "DOCUMENT"
+)
+
+// ProviderOnePassword is a provider for 1Password.
+type ProviderOnePassword struct {
+	vaults map[string]int
+	client connect.Client
+}
+
+// https://github.com/external-secrets/external-secrets/issues/644
+var _ esv1beta1.SecretsClient = &ProviderOnePassword{}
+var _ esv1beta1.Provider = &ProviderOnePassword{}
+
+// NewClient constructs a 1Password Provider.
+func (provider *ProviderOnePassword) NewClient(ctx context.Context, store esv1beta1.GenericStore, kube kclient.Client, namespace string) (esv1beta1.SecretsClient, error) {
+	config := store.GetSpec().Provider.OnePassword
+
+	credentialsSecret := &corev1.Secret{}
+	objectKey := types.NamespacedName{
+		Name:      config.Auth.SecretRef.ConnectToken.Name,
+		Namespace: namespace,
+	}
+
+	// only ClusterSecretStore is allowed to set namespace (and then it's required)
+	if store.GetObjectKind().GroupVersionKind().Kind == esv1beta1.ClusterSecretStoreKind {
+		if config.Auth.SecretRef.ConnectToken.Namespace == nil {
+			return nil, fmt.Errorf(errClusterSecretStoreNamespaceMissing)
+		}
+		objectKey.Namespace = *config.Auth.SecretRef.ConnectToken.Namespace
+	}
+
+	err := kube.Get(ctx, objectKey, credentialsSecret)
+	if err != nil {
+		return nil, fmt.Errorf(errFetchK8sSecret, err)
+	}
+	token := credentialsSecret.Data[config.Auth.SecretRef.ConnectToken.Key]
+	if (token == nil) || (len(token) == 0) {
+		return nil, fmt.Errorf(errMissingToken)
+	}
+	provider.client = connect.NewClientWithUserAgent(config.ConnectHost, string(token), userAgent)
+	provider.vaults = config.Vaults
+
+	return provider, nil
+}
+
+// ValidateStore checks if the provided store is valid.
+func (provider *ProviderOnePassword) ValidateStore(store esv1beta1.GenericStore) error {
+	return validateStore(store)
+}
+
+func validateStore(store esv1beta1.GenericStore) error {
+	storeSpec := store.GetSpec()
+	if storeSpec == nil || storeSpec.Provider == nil || storeSpec.Provider.OnePassword == nil {
+		return fmt.Errorf(errOnePasswordStore)
+	}
+	config := storeSpec.Provider.OnePassword
+
+	// only ClusterSecretStore is allowed to set namespace (and then it's required)
+	if store.GetObjectKind().GroupVersionKind().Kind == esv1beta1.ClusterSecretStoreKind {
+		if config.Auth.SecretRef.ConnectToken.Namespace == nil {
+			return fmt.Errorf(errClusterSecretStoreNamespaceMissing)
+		}
+	} else {
+		if config.Auth.SecretRef.ConnectToken.Namespace != nil {
+			return fmt.Errorf(errSecretStoreNamespaceSet)
+		}
+	}
```

Comment:
there is a method called`utils.ValidateSecretSelector` if I recall correctly... You can use it here for the `ConnectToken`

Reply:
Good call, done.